payu/experiment.py

```diff
@@ -769,8 +769,25 @@ def run(self, *user_flags):
         if run_script:
             self.run_userscript(run_script)
 
+    def archiving(self):
+        """
+        Determine whether to run archive step based on config.yaml settings.
+        Default to True when archive settings are absent.
+        """
+        archive_config = self.config.get('archive', {})
+        if isinstance(archive_config, dict):
+            return archive_config.get('enable', True)
+
+        # Backwards compatibility for configs with boolean archive setting
+        elif isinstance(archive_config, bool):
+            return archive_config
+
+        else:
+            msg = "Incorrect format for archive settings in config.yaml"
+            raise RuntimeError(msg)
+
```

Comment:
I've added this method assuming we'll swap the `archive` boolean to a dict in `config.yaml` files, e.g:
```yaml
archive:
    enable: True
    compress_logs: True
```
The above block will retain backwards compatibility with configs using an `archive: <True/False>` boolean`, and will also default to `True` when no `archive` settings are included.